packages/compass-data-modeling/src/utils/nodes-and-edges.tsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import toNS from 'mongodb-ns';
+import { InlineDefinition, Body, css } from '@mongodb-js/compass-components';
+import type { NodeProps, EdgeProps } from '@mongodb-js/diagramming';
+import type { MongoDBJSONSchema } from 'mongodb-schema';
+import type { SelectedItems } from '../store/diagram';
+import type {
+  DataModelCollection,
+  Relationship,
+} from '../services/data-model-storage';
+
+function getBsonTypeName(bsonType: string) {
+  switch (bsonType) {
+    case 'array':
+      return '[]';
+    default:
+      return bsonType;
+  }
+}
+
+const mixedTypeTooltipContentStyles = css({
+  overflowWrap: 'anywhere',
+  textWrap: 'wrap',
+  textAlign: 'left',
+});
+
+function getFieldTypeDisplay(bsonTypes: string[]) {
+  if (bsonTypes.length === 0) {
+    return 'unknown';
+  }
+
+  if (bsonTypes.length === 1) {
+    return getBsonTypeName(bsonTypes[0]);
+  }
+
+  const typesString = bsonTypes
+    .map((bsonType) => getBsonTypeName(bsonType))
+    .join(', ');
+
+  // We show `mixed` with a tooltip when multiple bsonTypes were found.
+  return (
+    <InlineDefinition
+      definition={
+        <Body className={mixedTypeTooltipContentStyles}>
+          Multiple types found in sample: {typesString}
+        </Body>
+      }
+    >
+      <div>(mixed)</div>
+    </InlineDefinition>
+  );
+}
+
+export const getSelectedFields = (
+  selectedItems: SelectedItems | null,
+  relationships?: Relationship[]
+): Record<string, string[] | null | undefined> => {
+  if (!selectedItems || selectedItems.type !== 'relationship') return {};
+  const { id } = selectedItems;
+  const { relationship } = relationships?.find((rel) => rel.id === id) ?? {};
+  const selection: Record<string, string[] | null | undefined> = {};
+  if (relationship?.[0].ns) {
+    selection[relationship[0].ns] = relationship[0].fields;
+  }
+  if (relationship?.[1].ns) {
+    selection[relationship[1].ns] = relationship[1].fields;
+  }
```

Comment:
One small drive-by change here. Before you had to have both namespaces selected to have either of the fields to start showing up in UI, I adjusted it a bit to make sure that field selections start to show up as soon as possible when you have both namespace and field set